Assets/MixedRealityToolkit.SDK/Features/Utilities/StabilizationPlaneModifier.cs

```diff
@@ -0,0 +1,390 @@
+            return false;
+        }
+
+        /// <summary>
+        /// Configures the stabilization plane to update its position based on an object in the scene.        
+        /// </summary>
+        private void ConfigureTransformOverridePlane(float deltaTime)
+        {
+            planePosition = TargetOverride.position;
+
+            Vector3 gazeOrigin = GazeOrigin;
+            Vector3 gazeDirection = TargetOverride.position - InputSystem.GazeProvider.GazePointer.Position;
+            Vector3 velocity = Vector3.zero;
+            if (TrackVelocity)
+            {
+                velocity = UpdateVelocity(deltaTime);
+            }
+
+            float focusPointDistance = gazeDirection.magnitude;
+            float lerpPower = focusPointDistance > currentPlaneDistance ? LerpStabilizationPlanePowerFarther
+                                                                        : LerpStabilizationPlanePowerCloser;
+
+            // Smoothly move the focus point from previous hit position to new position.
+            currentPlaneDistance = Mathf.Lerp(currentPlaneDistance, focusPointDistance, lerpPower * deltaTime);
+
+            gazeDirection.Normalize();
+            planePosition = gazeOrigin + (gazeDirection * currentPlaneDistance);
+
+#if UNITY_EDITOR
+            debugPlane.Center = planePosition;
+            debugPlane.Normal = -gazeDirection;
+#elif UNITY_WSA
+            // Place the plane at the desired depth in front of the user and billboard it to the gaze origin.
+            HolographicSettings.SetFocusPointForFrame(planePosition, -gazeDirection, velocity);
+#endif
+        }
+
+        private void ConfigureOverridePlane(float deltaTime)
+        {
+            planePosition = OverridePlane.Center;
+
+            Vector3 velocity = Vector3.zero;
+            if (TrackVelocity)
+            {
+                velocity = UpdateVelocity(deltaTime);
+            }
+
+            Vector3 gazeOrigin = GazeOrigin;
+            Vector3 gazeToPlane = planePosition - gazeOrigin;
+            float focusPointDistance = gazeToPlane.magnitude;
+            float lerpPower = focusPointDistance > currentPlaneDistance ? LerpStabilizationPlanePowerFarther
+                                                                        : LerpStabilizationPlanePowerCloser;
+
+            // Smoothly move the focus point from previous hit position to new position.
+            currentPlaneDistance = Mathf.Lerp(currentPlaneDistance, focusPointDistance, lerpPower * deltaTime);
+            gazeToPlane.Normalize();
+            planePosition = gazeOrigin + (gazeToPlane * currentPlaneDistance);
+
+#if UNITY_EDITOR
+            debugPlane.Center = planePosition;
+            debugPlane.Normal = OverridePlane.Normal;
+#elif UNITY_WSA
+            // Place the plane at the desired depth in front of the user and billboard it to the gaze origin.
+            HolographicSettings.SetFocusPointForFrame(planePosition, OverridePlane.Normal, velocity);
```

Comment:
Nit: Consider moving this code to a common helper (i.e. this code + the code in ConfigureTransformOverridePlane have a bunch of shared stuff that can probably be moved into a private helper)